src/Gui/Viewer/TrackballCameraManipulator.cpp

```diff
@@ -184,22 +159,25 @@ bool Gui::TrackballCameraManipulator::handleMouseMoveEvent(
 
     emit cameraChanged( m_camera->getPosition(), m_target );
 
-    return m_cameraRotateMode || m_cameraPanMode || m_cameraZoomMode;
+    return m_currentAction.isValid();
 }
 
 bool Gui::TrackballCameraManipulator::handleMouseReleaseEvent( QMouseEvent* /*event*/ ) {
-    m_cameraRotateMode    = false;
-    m_cameraPanMode       = false;
-    m_cameraZoomMode      = false;
+    m_currentAction       = Ra::Core::Utils::Index::Invalid();
     m_quickCameraModifier = 1.0_ra;
-
     return true;
 }
 
-bool Gui::TrackballCameraManipulator::handleWheelEvent( QWheelEvent* event ) {
+bool Gui::TrackballCameraManipulator::handleWheelEvent( QWheelEvent* event,
+                                                        const Qt::MouseButtons& buttons,
+                                                        const Qt::KeyboardModifiers& modifiers,
+                                                        int key
 
-    handleCameraZoom( ( event->angleDelta().y() * 0.01_ra + event->angleDelta().x() * 0.01_ra ) *
-                      m_wheelSpeedModifier );
+) {
+    ///\todo use action
```

Comment:
I do not understand what this todo means

Reply:
will be done ;)